### DISCUSSION TITLE
Dremel Idea Builder 3D20 Bed Leveling Issue

### DISCUSSION THREAD

**Discussion by Blitzpwnage** [moonglow/FlashForge_Marlin#94] [Q&A]
I just installed Marlin on my Idea Builder 3D20 and the bed leveling seems to be backwards. It does one point touch in the front then 2 in the back. This does not correspond to the adjustment knobs on the printer bed and is leader to some real troubleshooting when bed leveling. I can usually level the bed eventually but man is it a pain! Is there a fix for this in the works at all? 

Thanks!

**Comment by moonglow** [comment]
@Blitzpwnage  Hi there !

You need to use manual axis movement ( just click on position bar and it will open manual movement panel )  to locate each knob with your print head nozzle and tell me coordinate ( X, Y ) for each of them, i will add it to calibration script. I already ask about it, but not get answer yet

**Reply by Blitzpwnage** [reply]
I’ll get the coordinates for you this week! Glad to hear I can help out with this! 
Expect to see a reply from me in the next few days, thanks for the help 🙂

**Reply by Blitzpwnage** [reply]
@moonglow Hi there I have the coordinates for the bed leveling knobs, I included some pics hope they help!

Back middle knob X(-2) Y (66)
![19B82F35-FDB5-46B2-B9B3-566F237E38E5](https://user-images.githubusercontent.com/100648426/156933410-c6348b6d-6d52-4271-8e3c-ca7d6ae42616.jpeg)

Front left knob X(-52) Y(-64)
![2FC96B42-C124-49AA-B804-441F040471A9](https://user-images.githubusercontent.com/100648426/156933411-5a81a269-9625-4579-8cbe-979bf5b324f1.jpeg)

Front right knob X(49) Y (-64)
![98050B01-19D8-4C5A-88C2-0C2BF91D3CCD](https://user-images.githubusercontent.com/100648426/156933412-1b962d53-dc4c-497e-bc92-3d239a0e513f.jpeg)

Thanks!

**Reply by moonglow** [reply]
Thank you for point me ! I also tried to analyze original firmware and looks i found calibration script that was used by original Dremel firmware and original points looks like very near to yours! 

```G-Code
G1 X-57.7 Y-67 Z10 F3000
G1 Z0 F800
M0
G1 X41.3 Y-67 Z10 F3000
G1 Z0 F800
M0
G1 X-6.6 Y60 Z10 F3000
G1 Z0 F800
M0
G1 X0 Y0 Z10 F3000
G1 Z0 F800
M0
G28 X Y
G1 Z50 F800
```

Can you check that build ? if all be OK i will update my repo:
[dremel_2.0.9.3_03082022.zip](https://github.com/moonglow/FlashForge_Marlin/files/8204274/dremel_2.0.9.3_03082022.zip)

**Reply by Blitzpwnage** [reply]
@moonglow I will check it when I get a chance. I'll need to use the method below to push the update right? As I cannot push the update through a computer anymore, the Digilab Firmware Update method ? Or has this been solved?

How to Update Marlin firmware or revert to stock firmware:

Turn OFF your printer
Flip the printer upside-down and unscrew the six 2.5mm hex screws from the printer's bottom metal shield. Note that the bottom shield is held onto the printer via a metal grounding strap, so exercise caution when lifting it to not damage your printer.
The micro SD card is held in place via a small metal retaining clip. To remove internal micro SD card from the motherboard first note the position of the SD card in the slot so that you know how to place it back after step 4, then slide the metal clip toward the USB connector, then flip it up. Now the micro SD can be removed.
Using a computer, copy either the updated Marlin firmware or the original Dremel firmware onto the SD card's sys folder, and rename the firmware to dremel.bin
Insert back microSD card to motherboard socket ( be careful and make sure that the micro SD is seated correctly) then flip the metal locking bracket down and slide it away from the USB connector.
Turn ON the printer and wait until the Marlin firmware finishes booting. Then navigate to Settings->Configuration->Advanced Settings->Firmware update trigger
Accept the Write trigger? request when prompted.
Reboot your printer by turning it OFF then ON and it will update the firmware from SD card sys folder

**Reply by moonglow** [reply]
> As I cannot push the update through a computer anymore, the Digilab Firmware Update method ? Or has this been solved?

Original Dremel bootloader does not support fw update via itself ( dreamer bootloader allows it ) so  you can use that instruction from [WIKI](https://github.com/moonglow/FlashForge_Marlin/wiki/Installing-Marlin-on-Dremel-3D20) .

**Reply by ShinNinth** [reply]
As i was looking to move my 3d20 to This firmware i tested this build for you the leveling works perfectly with what you had posted here.  Hope this is the response you were looking for thanks!

**Reply by Blitzpwnage** [reply]
Sorry for the delay! Just wanted to give an update too. I was able to install the new build and am able to confirm that the bed leveling is now correct. Thanks again for the help @moonglow

**Reply by Blitzpwnage** [reply]
I started getting some thermal runaway issues but I don't think it's firmware related but am testing to see. Will update.

Edit: I reverted back to this firmware:https://github.com/moonglow/FlashForge_Marlin/releases/download/v0.14.2/color_ui_03012022.zip

I am no longer getting any thermal runaway issues. Not sure what I can provide to help or it was just me causing issues. I am using the same printer and settings as when the thermal runaway happened but with the reverted firmware and everything is going well so far. A bit strange. Sad because I have to bed level with the old way but I can still do it.
